### ISSUE TITLE
[Bug]: Parallel requests to Cohere and Anthropic return the same output content

### ISSUE THREAD

**Issue by ishaan-jaff** [BerriAI/litellm#413]
### What happened?

Made a parallel request to gpt-3.5, claude-2, command-nightly using litellm.completion. got the following response:
It looks like claude-2 and command-nightly had the exact same content. This needs further investigation - i'm suspicious this might be a bug 

```json
[<OpenAIObject chat.completion id=chatcmpl-80tEGyXDARuC1OSIyvV7tu9WHcnPw at 0x11b013600> JSON: {
  "id": "chatcmpl-80tEGyXDARuC1OSIyvV7tu9WHcnPw",
  "object": "chat.completion",
  "created": 1695222992,
  "model": "gpt-3.5-turbo-0613",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "In the realm of dreams, where poets dwell,\nI yearn to cast my words, to weave a spell.\nWith a quill in hand, my thoughts they roam,\nGuided by music, the rhythm finds its home.\n\nAcross the starlit canvas, my verses dance,\nPainting emotions in hues of circumstance.\nFrom whispered secrets and tales untold,\nTo melodies of love, both young and old.\n\nOh, behold the symphony of sky and earth,\nAs I search for meaning, for insight and worth.\nThrough ancient forests and oceans deep,\nTo mountaintops where dreams forever sleep.\n\nI'll spin a sonnet with threads of wonder,\nLike a gentle breeze on a summer's plunder.\nSeeking solace within the lines I etch,\nA refuge for hearts whose flames they fetch.\n\nWith pen poised lightly, I gather my thoughts,\nUnleashing a torrent, as time swiftly knots.\nEmbracing the silence, the words take flight,\nMirroring the beauty of a starry night.\n\nThrough verses, I'll capture ethereal scenes,\nA tapestry of emotions, dreams, and routines.\nFor poetry is a language beyond words,\nA sanctuary for hope, where freedom soars.\n\nSo let us, dear reader, embark on this quest,\nTo a place where emotions reign supreme and blessed.\nTogether we'll traverse this lyrical realm,\nAs I pen this poem, with heart at its helm."
      },
      "finish_reason": "stop"
    }
  ],
  "usage": {
    "prompt_tokens": 10,
    "completion_tokens": 285,
    "total_tokens": 295
  }
}, <ModelResponse chat.completion id=chatcmpl-6363db2f-dfba-4223-bcdf-bead94828124 at 0x11b030a40> JSON: {
  "object": "chat.completion",
  "choices": [
    {
      "finish_reason": "max_tokens",
      "index": 0,
      "message": {
        "content": " Sure! Here's a poem about the beauty of nature:\n\n\"Nature's beauty, so pure and true,\nA gift from the universe, for you.\n\nWith its vast and open skies,\nAnd the mountains that touch the highs.\n\nThe oceans so deep and blue,\nTheir depths, a mystery, we must pursue.\n\nThe forests so green and tall,\nWith the animals that roam, it calls.\n\nThe flowers so bright and bold,\nTheir scent, a fragrance, that always tells.\n\nNature's beauty, so pure and true,\nA gift from the universe, for you.\"",
        "role": "assistant",
        "logprobs": null
      }
    }
  ],
  "id": "chatcmpl-6363db2f-dfba-4223-bcdf-bead94828124",
  "created": 1695223009.931566,
  "model": "claude-instant-1.2",
  "usage": {
    "prompt_tokens": 11,
    "completion_tokens": 236,
    "total_tokens": 247
  }
}, <ModelResponse chat.completion id=chatcmpl-5407ce83-e465-493c-9041-f6cfd03ab56c at 0x108baf790> JSON: {
  "object": "chat.completion",
  "choices": [
    {
      "finish_reason": "stop",
      "index": 0,
      "message": {
        "content": " Sure! Here's a poem about the beauty of nature:\n\n\"Nature's beauty, so pure and true,\nA gift from the universe, for you.\n\nWith its vast and open skies,\nAnd the mountains that touch the highs.\n\nThe oceans so deep and blue,\nTheir depths, a mystery, we must pursue.\n\nThe forests so green and tall,\nWith the animals that roam, it calls.\n\nThe flowers so bright and bold,\nTheir scent, a fragrance, that always tells.\n\nNature's beauty, so pure and true,\nA gift from the universe, for you.\"",
        "role": "assistant",
        "logprobs": null
      }
    }
  ],
  "id": "chatcmpl-5407ce83-e465-493c-9041-f6cfd03ab56c",
  "created": 1695223013.816148,
  "model": "command-nightly",
  "usage": {
    "prompt_tokens": 3,
    "completion_tokens": 114,
    "total_tokens": 117
  }
```

**Comment by ghost** [comment]
This looks bad. We should prioritize this. I think I have context for this too, happy to pick up unless you're already on this @ishaan-jaff

**Comment by ishaan-jaff** [comment]
all yours @krrishdholakia - happy to take a look at this later today if you haven't had a chance to get to it by then

**Comment by ghost** [comment]
Found the issue. Should be resolved here - fc922bc3009370e291ae434bc14e956aa6137345. 

We should avoid initializing values in the function definitions, it gets overwritten in subsequent initializations

**Comment by ghost** [comment]
Closing as this is now in pypi - 0.1.720
